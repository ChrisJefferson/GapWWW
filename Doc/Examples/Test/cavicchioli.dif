33,35c33,286
< <b>. . .</b>
< #I  	254815  185	254630  254693
< #I  	255812  188	255624  255690
---
> #I  	2994	6	2988	2991
> #I  	3992	8	3984	3989
> #I  	4992	8	4984	4984
> #I  	5988	12	5976	5980
> #I  	6987	13	6974	6979
> #I  	7986	14	7972	7978
> #I  	8984	16	8968	8970
> #I  	9982	18	9964	9968
> #I  	10982	18	10964	10968
> #I  	11982	18	11964	11968
> #I  	12982	18	12964	12968
> #I  	13981	19	13962	13967
> #I  	14980	20	14960	14966
> #I  	15980	20	15960	15966
> #I  	16975	25	16950	16955
> #I  	17973	27	17946	17953
> #I  	18973	27	18946	18953
> #I  	19970	30	19940	19950
> #I  	20967	33	20934	20947
> #I  	21965	35	21930	21945
> #I  	22963	37	22926	22943
> #I  	23963	37	23926	23943
> #I  	24963	37	24926	24943
> #I  	25963	37	25926	25943
> #I  	26960	40	26920	26940
> #I  	27960	40	27920	27940
> #I  	28960	40	28920	28940
> #I  	29960	40	29920	29940
> #I  	30960	40	30920	30940
> #I  	31960	40	31920	31940
> #I  	32959	41	32918	32919
> #I  	33958	42	33916	33918
> #I  	34958	42	34916	34918
> #I  	35958	42	35916	35918
> #I  	36957	43	36914	36917
> #I  	37956	44	37912	37916
> #I  	38954	46	38908	38914
> #I  	39950	50	39900	39910
> #I  	40949	51	40898	40909
> #I  	41949	51	41898	41909
> #I  	42949	51	42898	42909
> #I  	43948	52	43896	43908
> #I  	44948	52	44896	44908
> #I  	45948	52	45896	45908
> #I  	46948	52	46896	46908
> #I  	47945	55	47890	47905
> #I  	48943	57	48886	48903
> #I  	49941	59	49882	49901
> #I  	50941	59	50882	50901
> #I  	51937	63	51874	51897
> #I  	52935	65	52870	52895
> #I  	53933	67	53866	53893
> #I  	54931	69	54862	54891
> #I  	55931	69	55862	55891
> #I  	56930	70	56860	56890
> #I  	57930	70	57860	57890
> #I  	58929	71	58858	58889
> #I  	59929	71	59858	59889
> #I  	60929	71	60858	60889
> #I  	61929	71	61858	61889
> #I  	62929	71	62858	62889
> #I  	63927	73	63854	63887
> #I  	64927	73	64854	64854
> #I  	65927	73	65854	65854
> #I  	66927	73	66854	66854
> #I  	67927	73	67854	67854
> #I  	68927	73	68854	68854
> #I  	69926	74	69852	69853
> #I  	70924	76	70848	70851
> #I  	71924	76	71848	71851
> #I  	72924	76	72848	72851
> #I  	73924	76	73848	73851
> #I  	74924	76	74848	74851
> #I  	75924	76	75848	75851
> #I  	76924	76	76848	76851
> #I  	77924	76	77848	77851
> #I  	78924	76	78848	78851
> #I  	79924	76	79848	79851
> #I  	80924	76	80848	80851
> #I  	81923	77	81846	81850
> #I  	82923	77	82846	82850
> #I  	83923	77	83846	83850
> #I  	84922	78	84844	84849
> #I  	85922	78	85844	85849
> #I  	86922	78	86844	86849
> #I  	87922	78	87844	87849
> #I  	88921	79	88842	88848
> #I  	89918	82	89836	89845
> #I  	90918	82	90836	90845
> #I  	91915	85	91830	91842
> #I  	92912	88	92824	92839
> #I  	93910	90	93820	93837
> #I  	94908	92	94816	94835
> #I  	95904	96	95808	95831
> #I  	96903	97	96806	96830
> #I  	97903	97	97806	97830
> #I  	98903	97	98806	98830
> #I  	99903	97	99806	99830
> #I  	100903	97	100806	100830
> #I  	101902	98	101804	101829
> #I  	102902	98	102804	102829
> #I  	103902	98	103804	103829
> #I  	104902	98	104804	104829
> #I  	105900	100	105800	105827
> #I  	106899	101	106798	106826
> #I  	107897	103	107794	107824
> #I  	108897	103	108794	108824
> #I  	109897	103	109794	109824
> #I  	110897	103	110794	110824
> #I  	111897	103	111794	111824
> #I  	112894	106	112788	112821
> #I  	113894	106	113788	113821
> #I  	114894	106	114788	114821
> #I  	115892	108	115784	115819
> #I  	116892	108	116784	116819
> #I  	117891	109	117782	117818
> #I  	118891	109	118782	118818
> #I  	119890	110	119780	119817
> #I  	120890	110	120780	120817
> #I  	121890	110	121780	121817
> #I  	122890	110	122780	122817
> #I  	123887	113	123774	123814
> #I  	124885	115	124770	124812
> #I  	125883	117	125766	125810
> #I  	126883	117	126766	126810
> #I  	127879	121	127758	127806
> #I  	128877	123	128754	128755
> #I  	129875	125	129750	129753
> #I  	130873	127	130746	130751
> #I  	131872	128	131744	131750
> #I  	132872	128	132744	132750
> #I  	133872	128	133744	133750
> #I  	134872	128	134744	134750
> #I  	135872	128	135744	135750
> #I  	136872	128	136744	136750
> #I  	137872	128	137744	137750
> #I  	138872	128	138744	138750
> #I  	139871	129	139742	139749
> #I  	140871	129	140742	140749
> #I  	141871	129	141742	141749
> #I  	142871	129	142742	142749
> #I  	143871	129	143742	143749
> #I  	144868	132	144736	144746
> #I  	145867	133	145734	145745
> #I  	146867	133	146734	146745
> #I  	147867	133	147734	147745
> #I  	148867	133	148734	148745
> #I  	149867	133	149734	149745
> #I  	150867	133	150734	150745
> #I  	151865	135	151730	151743
> #I  	152864	136	152728	152742
> #I  	153864	136	153728	153742
> #I  	154864	136	154728	154742
> #I  	155864	136	155728	155742
> #I  	156864	136	156728	156742
> #I  	157864	136	157728	157742
> #I  	158863	137	158726	158741
> #I  	159863	137	159726	159741
> #I  	160863	137	160726	160741
> #I  	161862	138	161724	161740
> #I  	162861	139	162722	162739
> #I  	163861	139	163722	163739
> #I  	164861	139	164722	164739
> #I  	165861	139	165722	165739
> #I  	166861	139	166722	166739
> #I  	167860	140	167720	167738
> #I  	168859	141	168718	168737
> #I  	169859	141	169718	169737
> #I  	170859	141	170718	170737
> #I  	171859	141	171718	171737
> #I  	172859	141	172718	172737
> #I  	173859	141	173718	173737
> #I  	174856	144	174712	174734
> #I  	175856	144	175712	175734
> #I  	176856	144	176712	176734
> #I  	177856	144	177712	177734
> #I  	178856	144	178712	178734
> #I  	179856	144	179712	179734
> #I  	180856	144	180712	180734
> #I  	181856	144	181712	181734
> #I  	182856	144	182712	182734
> #I  	183856	144	183712	183734
> #I  	184856	144	184712	184734
> #I  	185856	144	185712	185734
> #I  	186856	144	186712	186734
> #I  	187856	144	187712	187734
> #I  	188856	144	188712	188734
> #I  	189856	144	189712	189734
> #I  	190856	144	190712	190734
> #I  	191856	144	191712	191734
> #I  	192855	145	192710	192733
> #I  	193855	145	193710	193733
> #I  	194854	146	194708	194732
> #I  	195854	146	195708	195732
> #I  	196854	146	196708	196732
> #I  	197854	146	197708	197732
> #I  	198854	146	198708	198732
> #I  	199854	146	199708	199732
> #I  	200853	147	200706	200731
> #I  	201853	147	201706	201731
> #I  	202853	147	202706	202731
> #I  	203853	147	203706	203731
> #I  	204851	149	204702	204729
> #I  	205848	152	205696	205726
> #I  	206848	152	206696	206726
> #I  	207848	152	207696	207726
> #I  	208848	152	208696	208726
> #I  	209848	152	209696	209726
> #I  	210848	152	210696	210726
> #I  	211848	152	211696	211726
> #I  	212847	153	212694	212725
> #I  	213846	154	213692	213724
> #I  	214846	154	214692	214724
> #I  	215845	155	215690	215723
> #I  	216845	155	216690	216723
> #I  	217845	155	217690	217723
> #I  	218845	155	218690	218723
> #I  	219842	158	219684	219720
> #I  	220841	159	220682	220719
> #I  	221838	162	221676	221716
> #I  	222838	162	222676	222716
> #I  	223835	165	223670	223713
> #I  	224832	168	224664	224710
> #I  	225830	170	225660	225708
> #I  	226828	172	226656	226706
> #I  	227821	179	227642	227699
> #I  	228821	179	228642	228699
> #I  	229820	180	229640	229698
> #I  	230820	180	230640	230698
> #I  	231820	180	231640	231698
> #I  	232820	180	232640	232698
> #I  	233820	180	233640	233698
> #I  	234820	180	234640	234698
> #I  	235820	180	235640	235698
> #I  	236820	180	236640	236698
> #I  	237820	180	237640	237698
> #I  	238820	180	238640	238698
> #I  	239820	180	239640	239698
> #I  	240820	180	240640	240698
> #I  	241820	180	241640	241698
> #I  	242820	180	242640	242698
> #I  	243820	180	243640	243698
> #I  	244820	180	244640	244698
> #I  	245820	180	245640	245698
> #I  	246820	180	246640	246698
> #I  	247820	180	247640	247698
> #I  	248819	181	248638	248697
> #I  	249819	181	249638	249697
> #I  	250819	181	250638	250697
> #I  	251819	181	251638	251697
> #I  	252818	182	252636	252696
> #I  	253815	185	253630	253693
> #I  	254815	185	254630	254693
> #I  	255812	188	255624	255690
48a300
> #I  Options stack has been reset
