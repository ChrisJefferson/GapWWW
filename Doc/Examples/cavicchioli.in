f := FreeGroup(2); a:=f.1;; b:=f.2;;
h := f/[a*b*a^-2*b*a*b^-1,(b^-1*a^3*b^-1*a^-3)^1*a];
s := Subgroup(h,[h.1]);
SetInfoLevel(InfoFpGroup,3);
Index(h,s);
hom := FactorCosetAction(h,s);
p := Image(hom);
Size(p);
Index(h,TrivialSubgroup(h));
AbelianInvariants(h);
g := f/[a*b*a^-2*b*a*b^-1,(b^-1*a^3*b^-1*a^-3)^2*a];
# The following call is commented out because it causes a break loop
# It is demonstrated only in the interactive example
# Index(g,Subgroup(g,[g.1]));
SetInfoLevel(InfoFpGroup,0);
AbelianInvariants(g);
u := LowIndexSubgroupsFpGroup(g,TrivialSubgroup(g),15);;
u := Filtered(u,i->Index(g,i)>1);;
list := List(u,i->Image(FactorCosetAction(g,i)));;
Length(list);
id := List([1..24],
i->[Index(g,u[i]),Size(list[i]),TransitiveIdentification(list[i])]);;
Collected(id);
Collected(List(list,i->IsSimple(i)));
cores := List(u,i->Core(g,i));;
kernel := Intersection(cores);
hom := NaturalHomomorphismByNormalSubgroup(g,kernel);;
image := Image(hom);;
IsPermGroup(image);
NrMovedPoints(image);
Size(image);
List(u,i->AbelianInvariants(i));
quit;
