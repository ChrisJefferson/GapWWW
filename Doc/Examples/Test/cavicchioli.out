gap> f := FreeGroup(2); a:=f.1;; b:=f.2;;
<free group on the generators [ f1, f2 ]>
gap> h := f/[a*b*a^-2*b*a*b^-1,(b^-1*a^3*b^-1*a^-3)^1*a];
<fp group on the generators [ f1, f2 ]>
gap> s := Subgroup(h,[h.1]);
Group([ f1 ])
gap> SetInfoLevel(InfoFpGroup,3);
gap> Index(h,s);
#I  CosetTableFromGensAndRels called:
#I      defined deleted alive   maximal
#I  	13	1	12	13
12
gap> hom := FactorCosetAction(h,s);
[ f1, f2 ] -> [ (2,4,3,7,5)(6,8,10,11,9), (1,2,6,8,3)(5,9,12,10,7) ]
gap> p := Image(hom);
Group([ (2,4,3,7,5)(6,8,10,11,9), (1,2,6,8,3)(5,9,12,10,7) ])
gap> Size(p);
60
gap> Index(h,TrivialSubgroup(h));
#I  CosetTableFromGensAndRels called:
#I      defined deleted alive   maximal
#I  	133	13	120	133
120
gap> AbelianInvariants(h);
[  ]
gap> g := f/[a*b*a^-2*b*a*b^-1,(b^-1*a^3*b^-1*a^-3)^2*a];
<fp group on the generators [ f1, f2 ]>
gap> Index(g,Subgroup(g,[g.1]));
#I  CosetTableFromGensAndRels called:
#I      defined deleted alive   maximal
#I  	999	1	998	999
#I  	1997	3	1994	1996
<b>. . .</b>
#I  	254815  185	254630  254693
#I  	255812  188	255624  255690
Error, the coset enumeration has defined more than 256000 cosets
 called from
TCENUM.CosetTableFromGensAndRels( fgens, grels, fsgens ) called from
CosetTableFromGensAndRels( fgens, grels, fsgens ) called from
TryCosetTableInWholeGroup( H ) called from
CosetTableInWholeGroup( H ) called from
IndexInWholeGroup( H ) called from
...
Entering break read-eval-print loop ...
type 'return;' if you want to continue with a new limit of 512000 cosets,
type 'quit;' if you want to quit the coset enumeration,
type 'maxlimit := 0; return;' in order to continue without a limit
brk> quit;
gap> SetInfoLevel(InfoFpGroup,0);
gap> AbelianInvariants(g);
[  ]
gap> u := LowIndexSubgroupsFpGroup(g,TrivialSubgroup(g),15);;
gap> u := Filtered(u,i->Index(g,i)>1);;
gap> list := List(u,i->Image(FactorCosetAction(g,i)));;
gap> Length(list);
24
gap> id := List([1..24],
> i->[Index(g,u[i]),Size(list[i]),TransitiveIdentification(list[i])]);;
gap> Collected(id);
[ [ [ 11, 660, 5 ], 2 ], [ [ 11, 19958400, 7 ], 8 ], [ [ 12, 660, 179 ], 1 ], 
  [ [ 12, 95040, 295 ], 4 ], [ [ 12, 239500800, 300 ], 3 ], 
  [ [ 13, 3113510400, 8 ], 6 ] ]
gap> Collected(List(list,i->IsSimple(i)));
[ [ true, 24 ] ]
gap> cores := List(u,i->Core(g,i));;
gap> kernel := Intersection(cores);
Group(<fp, no generators known>)
gap> hom := NaturalHomomorphismByNormalSubgroup(g,kernel);;
gap> image := Image(hom);;
gap> IsPermGroup(image);
true
gap> NrMovedPoints(image);
237
gap> Size(image);
187838840781137083803280175646550961040094223793955611966199644315614527534543\
5097975617726355751505731109984424427520000000000000000000000000000000000000
gap> List(u,i->AbelianInvariants(i));
[ [ 11, 13 ], [ 2, 3, 9 ], [ 11, 13 ], [ 11, 13 ], [ 11, 13 ], [ 0, 0, 2 ], 
  [ 2, 2, 8 ], [ 2, 3, 9 ], [ 11, 13 ], [ 2, 2, 8 ], [ 11, 13 ], [ 2, 2, 8 ], 
  [ 2, 2, 8 ], [ 2, 2, 8 ], [ 2, 2, 3 ], [ 2, 2, 3 ], [ 2, 2, 3 ], 
  [ 2, 2, 3 ], [ 2, 2, 3 ], [ 2, 2, 3 ], [ 2, 2, 3 ], [ 2, 2, 3 ], 
  [ 2, 2, 3 ], [ 2, 2, 3 ] ]
gap> quit;
